pkgs/development/libraries/pcaudiolib/default.nix

```diff
@@ -1,33 +1,53 @@
-{ config, stdenv, lib, fetchFromGitHub
-, autoconf, automake, which, libtool, pkg-config
-, portaudio, alsa-lib
-, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux, libpulseaudio }:
+{ config
+, lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, libpulseaudio
+, libtool
+, pkg-config
+, portaudio
+, which
+, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
+}:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "pcaudiolib";
-  version = "1.1";
+  version = "1.2";
 
   src = fetchFromGitHub {
     owner = "espeak-ng";
-    repo = "pcaudiolib";
-    rev = version;
-    sha256 = "0c55hlqqh0m7bcb3nlgv1s4a22s5bgczr1cakjh3767rjb10khi0";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    hash = "sha256-ZG/HBk5DHaZP/H3M01vDr3M2nP9awwsPuKpwtalz3EE=";
```

Comment:
is finalAttrs really useful here? someone would need to override the version and hash anyway. Also, it's awkward if you do something like `pcaudiolib.overrideAttrs(oldAttrs: { pname = "pcausiolib_cmake"; nativeBuildInputs = oldAttrs.nativeBuilInputs ++ [ cmake ]; })`. Now the `src.repo` is wrong.

I guess I'm saying that trying to get rid of the `rec` isn't a compelling use  case of `finalAttrs`. Especially not for src, where src FOD is semi-independent of the logic on how package it.

Reply:
Well, my main motivation is getting rid of `rec`. On this regard, the new `finalAttrs` format is very useful.
In my opinion I believe this is not a sufficient reason to hinder such a migration, because it can be fixed one way or other.